packages/flutter/test/cupertino/date_picker_test.dart

```diff
@@ -2114,6 +2114,273 @@ void main() {
+    );
+
+    // For mode = CupertinoTimerPickerMode.ms
+    await tester.pumpWidget(
+      CupertinoApp(
+        home: Center(
+          child: CupertinoTimerPicker(
+            onTimerDurationChanged: (_) {},
+            mode: CupertinoTimerPickerMode.ms,
+            initialTimerDuration: const Duration(hours: 1, minutes: 1, seconds: 1),
+            selectionOverlayBuilder: (
+              BuildContext context,
+              int index,
+              int totalCount,
+            ) {
+              return selectionOverlay;
+            },
+          ),
+        ),
+      ),
+    );
+
+    // find the selection overlay
+    expect(find.byWidget(selectionOverlay), findsExactly(2));
+  });
+
+  testWidgets('CupertinoTimerPicker selectionOverlayBuilder with hm mode',
+      (WidgetTester tester) async {
+
+    final Widget selectionOverlay = Container(
+      color: const Color(0x12345678),
+    );
+
+    // For mode = CupertinoTimerPickerMode.hm
+    await tester.pumpWidget(
+      CupertinoApp(
+        home: Center(
+          child: CupertinoTimerPicker(
+            onTimerDurationChanged: (_) {},
+            mode: CupertinoTimerPickerMode.hm,
+            initialTimerDuration: const Duration(hours: 1, minutes: 1, seconds: 1),
+            selectionOverlayBuilder: (
+              BuildContext context,
+              int index,
+              int totalCount,
+            ) {
+              return selectionOverlay;
+            },
+          ),
+        ),
+      ),
+    );
+
+    // find the selection overlay
+    expect(find.byWidget(selectionOverlay), findsExactly(2));
+  });
+
+  testWidgets('CupertinoDatePicker selectionOverlayBuilder returns null',
+      (WidgetTester tester) async {
+
+    await tester.pumpWidget(
+      CupertinoApp(
+        home: Center(
+          child: CupertinoDatePicker(
+            onDateTimeChanged: (_) {},
+            initialDateTime: DateTime(2018, 9, 15),
+            selectionOverlayBuilder: (
+              BuildContext context,
+              int index,
+              int totalCount,
+            ) {
+              return null;
+            },
+          ),
+        ),
+      ),
+    );
+
+    expect(find.byType(CupertinoPicker), isNot(paints..rrect()));
+  });
+
+  testWidgets('CupertinoTimerPicker selectionOverlayBuilder returns null',
+      (WidgetTester tester) async {
+
+    await tester.pumpWidget(
+      CupertinoApp(
+        home: Center(
+          child: CupertinoTimerPicker(
+            onTimerDurationChanged: (_) {},
+            mode: CupertinoTimerPickerMode.hm,
+            initialTimerDuration: const Duration(hours: 1, minutes: 1, seconds: 1),
+            selectionOverlayBuilder: (
+              BuildContext context,
+              int index,
+              int totalCount,
+            ) {
+              return null;
+            },
+          ),
+        ),
+      ),
+    );
+
+    expect(find.byType(CupertinoPicker), isNot(paints..rrect()));
+  });
```

Comment:
What about the case where you explicitly pass `null` to selectionOverlayBuilder`, is that the same as if you don't pass it at all?

Reply:
should I test that for all the picker types?